src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/SqlParser.java

```diff
@@ -562,7 +565,9 @@ private void removeAliasPrefix(Where where, String alias) {
         if (where instanceof Condition) {
             Condition cond = (Condition) where;
             String aliasPrefix = alias + ".";
-            cond.setName(cond.getName().replaceFirst(aliasPrefix, ""));
+            if (cond.getName() != null) {
```

Comment:
Maybe I missed some case, but couldn't find one where name is not initialized.

Reply:
All the places where scripts and methods are part of **WHERE** clause. See below test cases. 

- com.amazon.opendistroforelasticsearch.sql.unittest.DateFunctionsTest.yearInWhere
 - com.amazon.opendistroforelasticsearch.sql.unittest.DateFunctionsTest.weekOfYearInWhere
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.numberEqualConditionWithoutProperty
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.whereConditionLeftFunctionRightPropertyGreatTest
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.whereConditionVariableRightVariableEqualTest
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.scriptOnFilterWithParams
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.stringAndNumberEqualConditionWithoutProperty
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.stringEqualConditionWithoutProperty
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.propertyGreatCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.scriptOnFilterNoParams
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.numberGreatConditionWithoutProperty
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.whereConditionLeftFunctionRightFunctionEqualTest
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.propertyEqualCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.parser.SqlParserTest.propertyWithTableAliasEqualCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.MathFunctionsTest.expm1WithValueArgument
 - com.amazon.opendistroforelasticsearch.sql.unittest.MathFunctionsTest.lowerCaseInWhere
 - com.amazon.opendistroforelasticsearch.sql.unittest.MathFunctionsTest.upperCaseInWhere
 - com.amazon.opendistroforelasticsearch.sql.unittest.MathFunctionsTest.expm1WithPropertyArgument
 - com.amazon.opendistroforelasticsearch.sql.unittest.MathFunctionsTest.eulersNumberInWhere
 - com.amazon.opendistroforelasticsearch.sql.unittest.MathFunctionsTest.piInWhere
 - com.amazon.opendistroforelasticsearch.sql.unittest.DateFormatTest.notEqualCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.DateFormatTest.simpleFormatCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.DateFormatTest.equalCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.DateFormatTest.greaterThanOrEqualToCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.DateFormatTest.greaterThanCondition
 - com.amazon.opendistroforelasticsearch.sql.unittest.DateFormatTest.timeZoneCondition